tasks/ThrustersInput.cpp

```diff
@@ -0,0 +1,192 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "ThrustersInput.hpp"
+
+using namespace auv_control;
+
+ThrustersInput::ThrustersInput(std::string const& name)
+    : ThrustersInputBase(name)
+{
+}
+
+ThrustersInput::ThrustersInput(std::string const& name, RTT::ExecutionEngine* engine)
+    : ThrustersInputBase(name, engine)
+{
+}
+
+ThrustersInput::~ThrustersInput()
+{
+}
+
+
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See ThrustersInput.hpp for more detailed
+// documentation about them.
+
+bool ThrustersInput::configureHook()
+{
+    if (! ThrustersInputBase::configureHook())
+        return false;
+
+    thrusterVoltage		= _thruster_voltage.get();
+    coeffPos 			= _thruster_coeff_pos.get();
+    coeffNeg 			= _thruster_coeff_neg.get();
+    controlModes		= _control_modes.get();
+
+    // Let's assume control_modes was set with the right number of thrusters
+    numberOfThrusters = controlModes.size();
+
+    // Those coefficients need to be set
+    if(coeffPos.empty() || coeffNeg.empty())
+    {
+		RTT::log(RTT::Error) << "Thruster coefficients were not set" << RTT::endlog();
+    	exception(WRONG_SIZE_OF_THRUSTER_COEFFICIENTS);
+    }
+
+
+    // Check size of parameters. They should met among them.
+    if(coeffPos.size() != numberOfThrusters || coeffNeg.size() != numberOfThrusters )
+    {
+    	if(coeffPos.size() != coeffNeg.size())
+    	{	// If coefficients sizes are different among them and controm_modes size, one of the coeff were not set right
+    		RTT::log(RTT::Error) << "Coefficients were not set with "<< numberOfThrusters <<" as predicted" << RTT::endlog();
+    		exception(WRONG_SIZE_OF_THRUSTER_COEFFICIENTS);
+    	}
+    	else if(!controlModes.empty())
+    	{
+    		// Well, if the thruster_coeff vectors have the same size, looks like the control_modes is the one with wrong size
+    		exception(WRONG_SIZE_OF_CONTROLMODES);
+    	}
+    	else if(controlModes.empty())
+    	{
+    		// In this case, control_modes is empty and coeffPos has the right number of thrusters
+    		// Let's set control_modes with default values RAW
+    		numberOfThrusters = coeffPos.size();
+        	//resize the controlModes to number of Thrusters
+        	controlModes.resize(numberOfThrusters);
+        	//set undefined controlModes to RAW
+        	for(unsigned int i = 0; i < numberOfThrusters; i++)
+        	{
+    			controlModes[i] = base::JointState::RAW;
+        	}
+    	}
+    }
+
+    // Check control_modes
+    for (int i=0; i<numberOfThrusters; i++)
+    {
+    	if(controlModes[i] !=  base::JointState::SPEED && controlModes[i] !=  base::JointState::RAW)
+    	{
+			RTT::log(RTT::Error) << "Control Mode should be SPEED or RAW" << RTT::endlog();
```

Comment:
Wrong indentation, you're mixing tabs and spaces and have your editor configured to do tab == 4 spaces.

Use tab==8 and indent with spaces.